src/typing/type_inference_js.ml

```diff
@@ -168,3 +168,39 @@ let infer_lib_file ~metadata ~exclude_syms file statements comments =
   ));
 
   cx, SMap.keys Scope.(module_scope.entries)
+
+let infer_resource_file ~metadata ~filename ~module_name =
+  Flow_js.Cache.clear();
+
+  let cx = Flow_js.fresh_context metadata filename module_name in
+  let exported_module_name = Modulename.to_string module_name in
+
+  let reason_exports_module =
+    Reason.reason_of_string (
+      Utils.spf "exports of module `%s`" exported_module_name) in
+  let reason_loc = Loc.({ none with source = Some filename }) in
+
+  let filename = Loc.(match filename with
+  | ResourceFile filename -> filename
+  | SourceFile _
+  | LibFile _
+  | Builtins
+  | JsonFile _ -> failwith "Resource file is not a resource file?!") in
+  let reason, exports_t = match Utils.extension_of_filename filename with
+  | Some ".css" ->
+      let reason = Reason.mk_reason
+        "Flow assumes requiring a .css file returns undefined"
```

Comment:
Maybe it makes sense to make `{ [key: string]: any }` the default since the people that will care about it will use it that way and the ones who don't won't use it at all.